gammapy/data/observations.py

```diff
@@ -194,6 +194,12 @@ def gti(self):
         gti = self.obs_filter.filter_gti(self._gti)
         return gti
 
+    @gti.setter
+    def gti(self, value):
```

Comment:
At the GADF DL3, the data should contain the GTI (see https://gamma-astro-data-formats.readthedocs.io/en/v0.3/general/hduclass.html). However, a science analysis tool can change the GTI for its internal purpose (see 
https://gamma-astro-data-formats.readthedocs.io/en/v0.3/events/gti.html).
So, having a setter makes sense.
However:
1. one should be sure very cautious to not araise existing GTI (for provenance)
2. one should create our own Gammapy time interval (or a `UserTimeInterval`) in order to not mix the concepts. This becomes more and more important.

I propose here to add in the code a test forbidding to erase existing DL3 GTI

Reply:
Maybe we can raise an error only if a GTI and the events exits because for simulation we can use flexibility in redefining the GTI.

Reply:
Indeed, when both exist, a error could be raised. This guaranties to not detroy DL3 data

Reply:
done